### PR TITLE
feat(client): signIn support for `redirectTo` and `resume`

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Ref https://github.com/mozilla/fxa-content-server/pull/3671
Ref https://github.com/mozilla/fxa-auth-server/pull/1232

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
I have to rebase this.

**Comment by shane-tomlinson** [comment]
@vladikoff, @vbudhram - can I get an r?

File: tests/lib/signIn.js [hunk 1]
```diff
@@ -88,6 +90,64 @@ define([
...
+          service: 'sync'
+        };
+
+        return respond(client.signIn(email, password, opts), RequestMocks.signIn)
+          .then(function (res) {
+            assert.ok(res.uid);
+            return respond(mail.wait(user), RequestMocks.mailServiceAndRedirect);
+          })
+          .then(
+            function (emails) {
```
Comment (vbudhram) [inline tests/lib/signIn.js]: nit: Not sure how the var name `emails` relates to the sign-in response?

**Comment by vbudhram** [comment]
Besides nit, LGTM 👍

File: tests/lib/signIn.js (hunk 1 above)
Comment (shane-tomlinson) [inline tests/lib/signIn.js]: That is actually the response to the mailer on line 106.

**Pull request merged**